src/NumberInput/NumberInput.tsx

```diff
@@ -91,28 +89,21 @@ export const NumberInput = forwardRef(function NumberInput(
     return Math.round(event * 100) / 100
   }
 
-  const handleStrictValue = (event: number): number => {
-    if (isInRange(event)) {
+  const formatCurrency = (event: string) => {
+    const decimalIndex = event.indexOf('.')
+    if (decimalIndex >= 0 && event.length > decimalIndex + 1) {
+      const fractionalString = event.substring(decimalIndex + 1).substring(0, 2)
+      return `${event.substring(0, decimalIndex)}.${fractionalString}`
+    } else {
```

Comment:
think using a number here for the event may make this formatting here easier to do - some comments here would also be helpful too

Reply:
The issue we had was that using number caused some weird behaviours, typing `20.0` would be stringified to `20`, and the currency formatting didn't prevent adding extra zeros (`20.1000000`) because as a number it's just `20.1`, so the setState/whatever function passed into the component wouldn't run. Using string fixes a lot of these issues but I know it's weird in a number input component

Maybe something to chat about in FE guild tomorrow ⚔️

Reply:
Could we maybe do that in a separate pr? This function is just to handle currencies really